src/manifest.ts

```diff
@@ -18,58 +22,77 @@ export class SlackManifest {
 
   export() {
     const def = this.definition;
-    const manifest = {
-      "_metadata": {
+    const manifest: ManifestSchema = {
+      _metadata: {
         // todo: is there a more idiomatic way of defining this? constant file?
-        "major_version": 2,
+        major_version: 2,
       },
-      "display_information": {
+      display_information: {
         background_color: def.backgroundColor,
         name: def.name,
         long_description: def.longDescription,
         short_description: def.description,
       },
       icon: def.icon,
-      "oauth_config": {
+      oauth_config: {
         scopes: {
           bot: this.ensureBotScopes(),
         },
       },
       features: {
         bot_user: {
-          display_name: def.displayName ||
-            def.name,
+          display_name: def.displayName || def.name,
         },
       },
-      "outgoing_domains": def.outgoingDomains || [],
-    } as ManifestSchema;
+      outgoing_domains: def.outgoingDomains || [],
+    };
 
     if (def.functions) {
-      manifest.functions = def.functions?.reduce((acc = {}, fn) => {
-        acc[fn.id] = fn.export();
-        return acc;
-      }, {} as ManifestSchema["functions"]);
+      manifest.functions = def.functions?.reduce<ManifestFunctionsSchema>(
+        (acc = {}, fn) => {
+          acc[fn.id] = fn.export();
+          return acc;
+        },
+        {},
+      );
     }
 
     if (def.workflows) {
-      manifest.workflows = def.workflows?.reduce((acc = {}, workflow) => {
-        acc[workflow.id] = workflow.export();
-        return acc;
-      }, {} as ManifestSchema["workflows"]);
+      manifest.workflows = def.workflows?.reduce<ManifestWorkflowsSchema>(
+        (acc = {}, workflow) => {
+          acc[workflow.id] = workflow.export();
+          return acc;
+        },
+        {},
+      );
     }
 
     if (def.types) {
-      manifest.types = def.types?.reduce((acc = {}, customType) => {
-        acc[customType.id] = customType.export();
-        return acc;
-      }, {} as ManifestSchema["types"]);
+      manifest.types = def.types?.reduce<ManifestCustomTypesSchema>(
+        (acc = {}, customType) => {
+          acc[customType.id] = customType.export();
+          return acc;
+        },
+        {},
+      );
     }
 
     if (def.datastores) {
-      manifest.datastores = def.datastores?.reduce((acc = {}, datastore) => {
-        acc[datastore.name] = datastore.export();
-        return acc;
-      }, {} as ManifestSchema["datastores"]);
+      manifest.datastores = def.datastores?.reduce<ManifestDataStoresSchema>(
+        (acc = {}, datastore) => {
+          acc[datastore.name] = datastore.export();
+          return acc;
+        },
+        {},
+      );
+    }
+
+    if (def.features?.appHome) {
```

Comment:
This means that `app_home` won't be set unless the developer specifies something at `features.appHome`